aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1624,6 +1624,11 @@ Tensor flatten(const Tensor& self, DimnameList dims, Dimname out_dim) {
   return native::flatten(self, *dims.begin(), *(dims.end() - 1), out_dim);
 }
 
+Tensor ravel(const Tensor& self) {
+  // Same as flatten() with start_dim as 0 and end_dim as -1.
+  return native::flatten(self, 0, -1);
```

Comment:
nit: `self.reshape(-1)` avoids some of the computation that goes on inside `flatten`.

Calling `native::flatten` instead of `at::flatten` inside of here means that we won't actually get autograd support. Using at::flatten (or reshape as pointed out above) fixes that problem. You can test this with something like:
```
x = torch.randn(3, 3, requires_grad=True)
out = x.ravel().sum()
out.backward()  # I think this throws an error with the current implementation
```

Reply:
You are right. It doesn't really need to go into `flatten`.
But, t's interesting that it doesn't throw any error and gradient for the tensor is correct when I used `native::flatten`.

Reply:
Oh, I see what happened now. native::flatten calls at::reshape, and at::reshape is supported by the autograd.